src/utils/pgwire/src/pg_protocol.rs

```diff
@@ -1205,3 +1230,53 @@ pub mod truncated_fmt {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_redact_parsable_sql() {
+        let sql = r"
+        create source temp (k bigint, v varchar) with (
+            connector = 'datagen',
+            v1 = 123,
+            v2 = 'with',
+            v3 = false,
+            v4 = '',
+        ) FORMAT plain ENCODE json (a='1',b='2')
+        ";
+        assert_eq!(redact_sql(sql), "CREATE SOURCE temp (k BIGINT, v CHARACTER VARYING) WITH (connector = [REDACTED], v1 = [REDACTED], v2 = [REDACTED], v3 = [REDACTED], v4 = [REDACTED]) FORMAT PLAIN ENCODE JSON (a = [REDACTED], b = [REDACTED])");
```

Comment:
Hmm, it might be too aggressive to me... Can we just redact some sensitive fields? For example, any fields that contain `"password"`

Reply:
> Can we just redact some sensitive fields? For example, any fields that contain "password"

By maintaining a blacklist of sql options in the kernel?

Reply:
Yeah, it's aaceptable. As a quick fix, we can do it in some quick & dirty way, such as checking whether `field_name.contains("password")` or something.

Reply:
fixed